TLM/TMPE.API/Traffic/Data/SpeedValue.cs

```diff
@@ -0,0 +1,156 @@
+namespace TrafficManager.API.Traffic.Data {
+    using System;
+    using JetBrains.Annotations;
+    using UnityEngine;
+
+    /// <summary>
+    /// Represents a speed value expressed in km/hour.
+    /// </summary>
+    [Serializable]
+    public struct KmphValue {
+        public KmphValue(ushort kmph) {
+            Kmph = kmph;
+        }
+
+        public override string ToString() {
+            return $"{Kmph:0.0} km/h";
+        }
+
+        public ushort Kmph { get; }
+    }
+
+    /// <summary>
+    /// Represents a speed value expressed in miles/hour.
+    /// </summary>
+    [Serializable]
+    public struct MphValue {
+        public MphValue(ushort mph) {
+            Mph = mph;
+        }
+
+        public override string ToString() {
+            return $"{Mph} MPH";
+        }
+
+        public ushort Mph { get; }
```

Comment:
should be `readonly` to indicate it's changed precisely once.

Reply:
It is a property with get only. The IDE is not very happy about me adding that keyword.

Reply:
Ah, looks like the compiler automatically does it which is probably why it doesn't need `readonly` keyword.

https://docs.microsoft.com/en-us/dotnet/csharp/properties#read-only

Reply:
`readonly` keyword should be applied to the structs whenever possible. I'd suggest to do it in this case. A struct with `get`-only properties is not immutable from the compiler point of view, so e.g. you cannot pass it using `in` (because you still can have getters that mutate the struct).

The compiler doesn't mark structs as `readonly` automatically.

Why is your IDE unhappy about it?

Reply:
tried adding to a get-only property